xonsh/tools.py

```diff
@@ -205,6 +216,28 @@ def __eq__(self, other):
         return all(map(operator.eq, self, other))
 
 
+class EnvPathDir(EnvPath):
```

Comment:
Sorry, just resuming this work. I'm still not quite clear how `BASH_COMPLETIONS` can just be a list. Right now, `is_env_path` and `str_to_env_path` are used to validate/convert entries. So if I change the `EnvPath` to only include directories, wouldn't `BASH_COMPLETIONS` fail?